### DISCUSSION TITLE
change isotp padding within the scripting interface

### DISCUSSION THREAD

**Discussion by akudlacek** [collin80/SavvyCAN#709] [Q&A] [answered]
I have a system that cares about the padding sent to it and it uses the OBD-II protocol, so iso-tp messages. I am able to send messages and they are padded with 0x00 but I would like to change that to 0xFF. Any way to do this from the scripting interface?

**Comment by collin80** [comment]
No, in fact currently the padding byte is always hard coded to 0 everywhere. That's certainly not correct behavior. If it were hard coded it should be either 0xAA or 0xCC not 0 or 0xFF. So, this will have to be fixed.

**Comment by collin80** [comment, marked as answer]
I added support for this to the QT6WIP build. Now you can put a line like this:

isotp.setPaddingByte(0xFF);

to make the padding byte FF instead. The default is now also 0xAA if nothing else is specified. As of yet the rest of the program does not provide any means of setting the padding byte.

**Reply by akudlacek** [reply]
Awesome you're on the ball! I will try it out. I was having problems with the latest build on Ubuntu but I will see if i can get past that.

**Comment by akudlacek** [comment]
With the latest build I am having a crashing issue when i try to load or save a script.  I also get some other errors on startup. Have any ideas?
I see this with Ubuntu 22.04.3 and 23.10.1

at startup
![image](https://github.com/collin80/SavvyCAN/assets/7029639/a49ca4c6-d6c6-446c-9e3c-fc77fc30af54)

when clicking load in scripting interface
![image](https://github.com/collin80/SavvyCAN/assets/7029639/4a77946a-4a2a-49bd-a5e0-8cceb1872b65)

**Reply by collin80** [reply]
I probably need to test this with Wayland to fix the issues. I use KDE because I absolutely HATE Gnome with a fiery passion. Totally can't stand it. So, I haven't been doing any testing against Gnome and Wayland. I am using Ubuntu 23.10 though. Just with KDE instead. I guess I'd better test with Gnome too to figure out what needs to be added to the AppImage to make it work.

**Reply by akudlacek** [reply]
That's what I needed to know! I just installed KDE and all is working now. Thanks for the help.

**Comment by akudlacek** [comment]
Can you set the padding byte for UDS as well?
